test/unit/spec/EnrollUser_spec.js

```diff
@@ -0,0 +1,157 @@
+      globalSuccessFn: successSpy
+    }, settings));
+    Util.registerRouter(router);
+    Util.mockRouterNavigate(router);
+    Util.mockJqueryCss();
+
+    router.refreshAuthState('dummy-token');
+    settings = {
+      router: router,
+      successSpy: successSpy,
+      form: new EnrollUserForm($sandbox),
+      ac: authClient,
+      setNextResponse: setNextResponse
+    };
+    if (resUnauthenticatedIdx) {
+      setNextResponse(resUnauthenticatedIdx);
+      return Expect.waitForPrimaryAuth(settings);
+    } else {
+      setNextResponse(resProfileRequiredUpdate);
+      return Expect.waitForEnrollUser(settings);
+    }
+  }
+
+  Expect.describe('Enroll User Form', function () {
+    itp('has the correct title on the enroll form', function () {
+      return setup().then(function (test) {
+        expect(test.form.formTitle().text()).toContain('Create Account');
+      });
+    });
+    itp('has the correct title on the register button', function () {
+      return setup().then(function (test) {
+        expect(test.form.formButton()[0].value).toEqual('Register');
+      });
+    });
+    itp('does not allow empty form submit', function () {
+      return setup().then(function (test) {
+        test.form.formButton().click();
+        expect(test.form.errorMessage()).toEqual('We found some errors. Please review the form and make corrections.');
+      });
+    });
+    itp('renders the right fields based on API response', function () {
+      return setup().then(function (test) {
+        expect(test.form.formInputs('streetAddress').length).toEqual(1);
+        expect(test.form.formInputs('streetAddress').find('input').attr('placeholder')).toEqual('enter streetAddress *'); 
+        expect(test.form.formInputs('streetAddress').hasClass('okta-form-input-field input-fix')).toBe(true);
+
+        expect(test.form.formInputs('employeeId').length).toEqual(1);
+        expect(test.form.formInputs('employeeId').find('input').attr('placeholder')).toEqual('enter employeeId *'); 
+        expect(test.form.formInputs('employeeId').hasClass('okta-form-input-field input-fix')).toBe(true);
+      });
+    });
+
+    itp('makes call to enroll if isEnrollWithLoginIntent is true and then renders the right fields based on API response', function () {
+      return setup(null, resUnauthenticatedIdx).then(function (test) {
+        test.setNextResponse(resProfileRequiredNew);
+        test.form.$('.registration-link').click();
+        return Expect.waitForEnrollUser(test);
+      })
+        .then(function (test) {
+          expect(test.form.formInputs('streetAddress').length).toEqual(1);
+          expect(test.form.formInputs('streetAddress').hasClass('okta-form-input-field input-fix')).toBe(true);
+          expect(test.form.formInputs('employeeId').length).toEqual(1);
+          expect(test.form.formInputs('employeeId').hasClass('okta-form-input-field input-fix')).toBe(true);
+          return Expect.waitForEnrollUser(test);
```

Comment:
why do we need this `wait`? I don't see any submit form happening or something that will required a wait.

Reply:
removed